File: schemaorg-rs/src/service_channel.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type ServiceChannelAdditionalTypeFieldEnum = String;
///<https://schema.org/ServiceChannel>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct ServiceChannel {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/availableLanguage>
    #[serde(rename = "availableLanguage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub available_language: Vec<ServiceChannelAvailableLanguageFieldEnum>,
    ///<https://schema.org/servicePhone>
    #[serde(rename = "servicePhone")]
    #[serde_as(as = "OneOrMany<_>")]
    pub service_phone: Vec<ContactPoint>,
    ///<https://schema.org/processingTime>
    #[serde(rename = "processingTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub processing_time: Vec<Duration>,
    ///<https://schema.org/serviceLocation>
    #[serde(rename = "serviceLocation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub service_location: Vec<Place>,
    ///<https://schema.org/servicePostalAddress>
    #[serde(rename = "servicePostalAddress")]
    #[serde_as(as = "OneOrMany<_>")]
    pub service_postal_address: Vec<PostalAddress>,
    ///<https://schema.org/serviceUrl>
    #[serde(rename = "serviceUrl")]
    #[serde_as(as = "OneOrMany<_>")]
    pub service_url: Vec<String>,
    ///<https://schema.org/serviceSmsNumber>
    #[serde(rename = "serviceSmsNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub service_sms_number: Vec<ContactPoint>,
    ///<https://schema.org/providesService>
    #[serde(rename = "providesService")]
    #[serde_as(as = "OneOrMany<_>")]
    pub provides_service: Vec<Service>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<ServiceChannelMainEntityOfPageFieldEnum>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<ServiceChannelIdentifierFieldEnum>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<ServiceChannelDescriptionFieldEnum>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<ServiceChannelSubjectOfFieldEnum>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<ServiceChannelAdditionalTypeFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<ServiceChannelImageFieldEnum>,
}

File: schemaorg-ts/src/service_channel.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type ServiceChannelAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/ServiceChannel>
 **/
export interface ServiceChannel {
  "@context": string;
  "@type": "ServiceChannel";

  /**
   * <https://schema.org/availableLanguage>
   **/
  availableLanguage: all.OneOrMany<all.ServiceChannelAvailableLanguageFieldEnum>;

  /**
   * <https://schema.org/servicePhone>
   **/
  servicePhone: all.OneOrMany<all.ContactPoint>;

  /**
   * <https://schema.org/processingTime>
   **/
  processingTime: all.OneOrMany<all.Duration>;

  /**
   * <https://schema.org/serviceLocation>
   **/
  serviceLocation: all.OneOrMany<all.Place>;

  /**
   * <https://schema.org/servicePostalAddress>
   **/
  servicePostalAddress: all.OneOrMany<all.PostalAddress>;

  /**
   * <https://schema.org/serviceUrl>
   **/
  serviceUrl: all.OneOrMany<string>;

  /**
   * <https://schema.org/serviceSmsNumber>
   **/
  serviceSmsNumber: all.OneOrMany<all.ContactPoint>;

  /**
   * <https://schema.org/providesService>
   **/
  providesService: all.OneOrMany<all.Service>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.ServiceChannelMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.ServiceChannelIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.ServiceChannelDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.ServiceChannelSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.ServiceChannelAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.ServiceChannelImageFieldEnum>;
}
